src/serialization/pkl2proto.py

```diff
@@ -0,0 +1,154 @@
+import pickle
+import sys
+import numpy as np
+import font_pb2
+import serialization
+
+
+def parseDict(d, filename="No filename provided"):
+    '''
+    Read a Dict containing font curves and return the info in a standardized np array
+
+    Parameters
+    ----------
+    d: Python Dictionary with key of the chars and the value of a closed curves dict, 
+    the closed curve dict has keys of ints up to max num closed curves and lists
+    of assosiated bez curves. Each bez curve in the list of bez curves has 3 coordinates
+    which are x and y coords of the control points. 
+    
+    filename: Optional parameter for knowing where the reading of the data failed
+    
+    
+    Returns
+    -------
+    New NP array of size len(chars), maxClosedCurves, maxCurvePerGlyph, 3, 2
+    '''
+
+    chars = [
+        'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N',
+        'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b',
+        'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p',
+        'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'zero', 'one', 'two',
+        'three', 'four', 'five', 'six', 'seven', 'eight', 'nine', 'exclam',
+        'numbersign', 'dollar', 'percent', 'ampersand', 'asterisk', 'question',
+        'at'
+    ]
+
+    # print( set(d.keys()) - set(chars))
+
+    maxCurvePerGlyph = 80
+    maxClosedCurves = 20
+
+    # initialize the np array with all ones then multiply by 99
+    final = np.ones((len(chars), maxClosedCurves, maxCurvePerGlyph, 3, 2))
+    final = final * 999
+
+    for c in range(len(chars)):
+        # This will run per glyph
+        try:
+            # getting number of closed curves in this glyph
+
+            if list((d[chars[c]]).keys()) == []:
+                print("Glyph :", chars[c], "has no curves")
+                continue
+            numClosedCurves = max(
+                d[chars[c]].keys()) + 1  #add one bc its indexing at 0
+
+            # getting number of bez curves in this glyph
+            curvesInGlyph = 0
+            for counter in d[
+                    chars[c]]:  #counting number of bezcurves per glyph
+                curvesInGlyph += len(d[chars[c]][counter])
+            # use sum to check if there are too many curves or closed curves defining a glyph
+            if (curvesInGlyph < maxCurvePerGlyph) and (numClosedCurves <
+                                                       maxClosedCurves):
+                for counter in d[chars[c]]:
+                    # this runs for every closed curve
+
+                    for curve in range(len(d[chars[c]][counter])):
```

Comment:
see above `for curve in ...`

Reply:
The reason I used this is because I want the integer c so i can index the np array "final" within the same for loop.